feat: Add result handler for PostgreSQL (#1480)
GoogleCloudPlatform/professional-services-data-validator: docs/installation.md

The tool natively supports BigQuery connections. If you need to connect to other databases such as Teradata or Oracle, you will need to install the appropriate connection libraries. (See the [Connections](connections.md) page for details)

This tool can be installed locally on your machine or can be containerized and run with Docker.

## Prerequisites

- Any machine with Python 3.8+ installed. (Note: Support for Python 3.7 is deprecated since [PR #894](https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/894))

## Deploy Data Validation CLI on your machine

The Data Validation tooling requires Python 3.8+.

```
sudo apt-get install python3
sudo apt-get install python3-dev
```

Create and activate a new virtual environment to sandbox the tool and its
dependencies from your system installation of Python.

```

> python3.10 -m venv venv

Reviewer: ```suggestion
python3 -m venv venv
```

source venv/bin/activate
```

> Update pip and make sure gcc is installed in your environment.

Reviewer: ```suggestion
Update `pip` and make sure `gcc` is installed in your environment.
```

```
sudo apt-get update  && sudo apt-get install gcc -y && sudo apt-get clean
pip install --upgrade pip
```

> You can install the tool via [pip](https://pypi.org/project/google-pso-data-validator/1.2.0/).

Reviewer: ```suggestion
You can install the tool via [pip](https://pypi.org/project/google-pso-data-validator/#history).
```

  terraform/results_schema.json
```

#### Cloud Resource Creation - After success

You should see a dataset named `pso_data_validator` and a table named
`results` created inside of your project.

> You are now be ready to run data validation commands and output the results to BigQuery. See an example [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/docs/examples.md#store-results-in-a-bigquery-table).

Reviewer: ```suggestion
You are now ready to run data validation commands and output the results to BigQuery. See an example [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/docs/examples.md#store-results-in-a-bigquery-table).
```

Author: Thanks for being diligent here, I obviously didn't proof read the text I was copying.

### PostgreSQL

In order to allow the data validation tool to write to a PostgreSQL table you must create a connection configuration to a PostgreSQL instance that authenticates as a user with privileges to write to a results table and optionally privileges to create the table.

The example below creates a user to create and update a results table in a schema named `pso_data_validator`:
```sql

> CREATE USER

Reviewer: ```suggestion
```
This row might have be duplicated by mistake.

Author: Good spot. Thanks

CREATE USER dvt_results_writer WITH PASSWORD 'S3cr3t!';
CREATE SCHEMA pso_data_validator;
GRANT CREATE ON SCHEMA pso_data_validator TO dvt_results_writer;
GRANT USAGE ON SCHEMA pso_data_validator TO dvt_results_writer;
```

The table will be created when first referenced in a result handler and appended to from there on.

Details on creating a PostgreSQL connection configuration can be found [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/docs/connections.md#Postgres).

> You are now be ready to run data validation commands and output the results to PostgreSQL.

Reviewer: ```suggestion
You are now ready to run data validation commands and output the results to PostgreSQL.
```
